### DISCUSSION TITLE
Sometimes malformed json with constrained decoding

### DISCUSSION THREAD

**Discussion by nivibilla** [sgl-project/sglang#240] [General]
When using a regex to constrain the output to a particular json format, sometimes after the comma it will randomly fail and do new lines. 

Like

```
{\n
    "tool" : "search",\n
    "query" : "something",\n
\n
\n
\n
...
}
```

**Comment by keepitsane** [comment]
I am also running into a similar issue when using the pydantic method for json decoding.

**Comment by nivibilla** [comment]
https://github.com/outlines-dev/outlines/issues/691

This issue seems to be talked about here too, they suggest

> In the build_json_from_object try adding `whitespace_pattern="" `or `whitespace_pattern="[ \n\t]?"`

Haven't tested though

**Comment by comaniac** [comment]
It's likely due to the regex issue. I'd suggest the following ways:
- Use a regex string directly instead of pydantic. If there are issues, you could post the `regex` here for help.
- If you really need to use pydantic/outlines, manually print the regex string and try to understand what it does. As my personal experience, the regex string generated by pydantic/outlines is not always correct or robust.

**Comment by nivibilla** [comment]
@comaniac thanks for your insight. It's interesting that the pydantic doesn't give perfect regex. Will keep that in mind. But I think also the solution mentioned in that other issue might be useful to test as well.

**Comment by nivibilla** [comment]
Could you convert this to a discussion since it's probably not an sglang issue
